From 9734d67cbc9ff4b62d75586ec5fbc2b1464dbf41 Mon Sep 17 00:00:00 2001
From: Robert Chiras <robert.chiras@nxp.com>
Date: Thu, 23 Nov 2017 14:31:43 +0200
Subject: [PATCH 2893/5242] MLK-16918: arm64: dts: fsl-imx8mq-evk: Fix dsi
 conflicts

commit  c6306557a19cd84ebdb756be00738e1542355bc3 from
https://source.codeaurora.org/external/imx/linux-imx.git

The MIPI-DSI nodes related to ADV7535 are in conflict with OV5640
MIPI-CSI camera, since they share the same address on the same i2c bus:
0x3c. In order to make DSI work with ADV7535 adapter, the OV5640 camera
needs to be disabled.
Also, for LCDIF (MXSFB), the irqsteer_dcss module needs to be disabled,
otherwise it will interfere in the suspend/resume routines.

Signed-off-by: Robert Chiras <robert.chiras@nxp.com>
Reviewed-by: Laurentiu Palcu <laurentiu.palcu@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 .../dts/freescale/fsl-imx8mq-evk-lcdif-adv7535.dts |    8 ++++++++
 .../dts/freescale/fsl-imx8mq-evk-lcdif-rm67191.dts |    4 ++++
 2 files changed, 12 insertions(+)

diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8mq-evk-lcdif-adv7535.dts b/arch/arm64/boot/dts/freescale/fsl-imx8mq-evk-lcdif-adv7535.dts
index fe34217..81ee650 100644
--- a/arch/arm64/boot/dts/freescale/fsl-imx8mq-evk-lcdif-adv7535.dts
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8mq-evk-lcdif-adv7535.dts
@@ -20,6 +20,14 @@
 	};
 };
 
+&irqsteer_dcss {
+	status = "disabled";
+};
+
+&ov5640_mipi {
+	status = "disabled";
+};
+
 &dcss_drm {
 	status = "disabled";
 };
diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8mq-evk-lcdif-rm67191.dts b/arch/arm64/boot/dts/freescale/fsl-imx8mq-evk-lcdif-rm67191.dts
index 90c33bd..1ae7d2b 100644
--- a/arch/arm64/boot/dts/freescale/fsl-imx8mq-evk-lcdif-rm67191.dts
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8mq-evk-lcdif-rm67191.dts
@@ -20,6 +20,10 @@
 	};
 };
 
+&irqsteer_dcss {
+	status = "disabled";
+};
+
 &dcss_drm {
 	status = "disabled";
 };
-- 
1.7.9.5

